packages/react-native/Libraries/SwiftExtensions/RCTWindow.swift

```diff
@@ -0,0 +1,37 @@
+import SwiftUI
+
+/**
+  `RCTWindow` is a SwiftUI struct that returns additional scenes.
+ 
+ Example usage:
+ ```
+ RCTWindow(id: "SecondWindow", sceneData: reactContext.getSceneData(id: "SecondWindow"))
+ ```
+ */
+public struct RCTWindow : Scene {
+  var id: String
+  var sceneData: RCTSceneData?
+  var moduleName: String
+  
+  public init(id: String, moduleName: String, sceneData: RCTSceneData?) {
+    self.id = id
+    self.moduleName = moduleName
+    self.sceneData = sceneData
+  }
+  
+  public var body: some Scene {
+    WindowGroup(id: id) {
+      if let sceneData {
+        RCTRootViewRepresentable(moduleName: moduleName, initialProps: sceneData.props)
+      }
```

Comment:
Let's throw an error/show warning here. If scene data is `nil` something went wrong

Reply:
Think ahead to how this case would be reported via crash reporting (eg Sentry). What information should be included in the exception message so it's actionable? Can the error be thrown/logged so the stack trace is cross-language, across JS and Swift?